client/src/components/atoms/ImagePicker.tsx

```diff
@@ -1,24 +1,167 @@
-import React, { FunctionComponent } from "react";
+import React, { FunctionComponent, useState } from "react";
+import Cropper from "react-easy-crop";
+import Dropzone from "react-dropzone";
 import ImageList from "./ImageList";
+import Slider from "@material-ui/core/Slider";
+
+import WarningBox from "./WarningBox";
 
 interface Props {
     onImageChange(url: string): void;
+    onUploadImg(data: string): void;
     images: string[];
     selected: string;
     isErroneous: boolean;
+    uploadedImg: string;
 }
 
+interface Dimensions {
+    width: number;
+    height: number;
+}
 const ImagePicker: FunctionComponent<Props> = (props: Props) => {
-    const { selected, images, onImageChange, isErroneous } = props;
+    const [crop, setCrop] = useState({ x: 0, y: 0 });
+    const [zoom, setZoom] = useState(1);
+    const [showWarning, setShowWarning] = useState(false);
+    const [warningText, setWarningText] = useState("");
+    const { selected, images, onImageChange, isErroneous, onUploadImg, uploadedImg } = props;
```

Comment:
nit: not a fan of destructuring props when props are just values (not objects, etc.) as it becomes more difficult to tell what is a prop and what isn't

Reply:
Updated.